src/ol/renderer/Composite.js

```diff
@@ -123,8 +127,13 @@ class CompositeMapRenderer extends MapRenderer {
         this.children_.push(element);
         previousElement = element;
       }
+      if ('getDeclutter' in layer) {
```

Comment:
is a `&& layer.getDeclutter()` missing?

Reply:
I did not add this intentionally, because I think in the future we'll want to add per-style decluttering. So you could set `declutter: true` on a `Style` and not have it set on the layer.

However, this question brings us back to the question where to create the `frameState`'s `declutterTree`. Because the way it is now we always create it as soon as a layer is capable of decluttering.

Reply:
Ah, I see. This code in `ol/renderer/canvas/VectorLayer` will not continue because `this.declutterExecutorGroup` will only be set inside of `prepareFrame()` if `layer.getDeclutter()` is true.
```
  renderDeclutter(frameState) {
    if (this.declutterExecutorGroup) {
      this.renderWorlds(
        this.declutterExecutorGroup,
        frameState,
        frameState.declutterTree
      );
    }
  }
```
Is it a real problem the the tree is created, though?

Reply:
It is not much of a problem, it just adds a little bit to the garbage that needs to be collection.